### DISCUSSION TITLE
Prisma MongoDB Views?

### DISCUSSION THREAD

**Discussion by ganesshkumar** [prisma/prisma#12021] [Q&A] [answered]
I have a MondoDB collection from which I have created a view using `MongoDB Compass` tool. 

Is it possible to query the data from this view using Prisma? I don't find any documentatin around how to update the schema.prisma file, so the generated PrismaClient can access this view with type safety.

**Comment by janpio** [comment, marked as answer]
Hi @ganesshkumar. MongoDB views are currently not supported by Prisma out of the box, as you probably figured out via `db pull` and them not showing up. 

For relational databases we have some workarounds (basically: just write a model manually that has the form the view has, and then it kinda works) but I am not sure if anyone has tried that yet with MongoDB. If you do, please let us know if it works as well so we can document that.

I would assume we will also support them when we get to https://github.com/prisma/prisma/issues/678 It can not hurt to comment about missing MongoDB views in that issue, or even creating a new standalone one. Then it would be easier to also link some documentation how views work on MongoDB and similar.

**Comment by ganesshkumar** [comment]
> For relational databases we have some workarounds (basically: just write a model manually that has the form the view has, and then it kinda works) but I am not sure if anyone has tried that yet with MongoDB. If you do, please let us know if it works as well so we can document that.

Yes, I tried creating a model for the view in the schema file and it works fine 👍

**Reply by janpio** [reply]
Thanks for letting me know. I opened a docs issue so we add this to our docs: https://github.com/prisma/docs/issues/2923

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
